### PR TITLE
Fixed drawing in BCEfficiencyFitter

### PR DESCRIPTION
**Pull request by greenwald**
Fixes issue #233 .

However, I have a segfault when I run the example `efficiencyFitterExample.C` that stems from the final command `hef.PrintAllMarginalized("distributions.pdf")`.

@fredRos, can you please check if you get this same segfault (also with my updated code in this PR)?

### CODE REVIEW

**Comment by fredRos** [comment]
You were running with root 6 I guess? I tried with root5.34.30. First I hit this CINT limitation

```
Error: i already declared as different type /home/beaujean/workspace/bat/build_247/examples/basic/efficiencyFitter/./../../../../examples/basic/efficiencyFitter/efficiencyFitterExample.C:88:
```
Upon changing `unsigned` to `int` in `for (int i = 0; i < 1000 ; ++i)`, the test passes and the output looks much better than before

[fit.pdf](https://github.com/bat/bat/files/1878977/fit.pdf)

**Comment by fredRos** [comment]
Could you please post the stacktrace of the segfault?

**Comment by greenwald** [comment]
I looked a little closer and saw that it had something to do with a mismatch of ROOT versions. (Though I had reconfigured and recompiled and reinstalled BAT with the same ROOT version. So I don't know why that should have been an issue. It was resolved with `make clean` and a repeat of the config/compile/install action.)

**Comment by fredRos** [comment]
> mismatch of ROOT versions.

good, I'm relieved. I have been working hard to remove any sources of segfaults from within BAT source code.

File: models/base/BCEfficiencyFitter.cxx [hunk 1]
```diff
@@ -137,22 +139,82 @@ void BCEfficiencyFitter::Fit()
...
+        legend->SetNColumns(2);
+        legend->SetColumnSeparation(5e-2);
+
+        legend->AddEntry(&fTrials, fTrials.GetTitle(), "F");
+        legend->AddEntry(&fSuccesses, fSuccesses.GetTitle(), "P");
+
+        legend->Draw();
+    }
+}
+
```
Comment (fredRos) [inline models/base/BCEfficiencyFitter.cxx]: Looks good

**Pull request merged**